src/main/java/com/zendesk/maxwell/schema/columndef/TimeColumnDef.java

```diff
@@ -1,27 +1,29 @@
 package com.zendesk.maxwell.schema.columndef;
 
 import java.sql.Time;
+import java.sql.Timestamp;
 
 import com.google.code.or.common.util.MySQLConstants;
 
-public class TimeColumnDef extends ColumnDef {
-	public TimeColumnDef(String name, String type, int pos) {
-		super(name, type, pos);
+public class TimeColumnDef extends ColumnDefWithLength {
+	public TimeColumnDef(String name, String type, int pos, Long columnLength) {
+		super(name, type, pos, columnLength);
 	}
 
 	@Override
 	public boolean matchesMysqlType(int type) {
-		return type == MySQLConstants.TYPE_TIME;
+		return type == MySQLConstants.TYPE_TIME || type == MySQLConstants.TYPE_TIME2;
 	}
 
-	@Override
-	public String toSQL(Object value) {
-		Time t = (Time) value;
-		return "'" + String.valueOf(t) + "'";
-	}
+	protected String formatValue(Object value) {
+		if ( value instanceof Timestamp ) {
+			Time time = new java.sql.Time(((Timestamp) value).getTime());
```

Comment:
no need to say `new java.sql.Time` -- just `new Time` will work since you've imported it